core-java/src/main/java/eu/neverblink/jelly/core/JellyConverterFactory.java

```diff
@@ -0,0 +1,121 @@
+package eu.neverblink.jelly.core;
+
+import eu.neverblink.jelly.core.internal.ProtoDecoderImpl;
+import eu.neverblink.jelly.core.internal.ProtoEncoderImpl;
+import eu.neverblink.jelly.core.proto.v1.RdfStreamOptions;
+
+/**
+ * "Main" interface to be implemented by RDF conversion modules (e.g., for Jena and RDF4J).
+ * Exposes factory methods for building protobuf encoders and decoders.
+ * <p>
+ * This should typically be implemented as an object. You should also provide a package-scoped given for your
+ * implementation so that users can easily make use of the connector in the stream package.
+ *
+ * @param <TNode> Type of RDF nodes in the RDF library
+ * @param <TDatatype> Type of RDF datatypes in the RDF library
+ * @param <TEncoderConverter> Implementation of ProtoEncoderConverter for a given RDF library.
+ * @param <TDecoderConverter> Implementation of ProtoDecoderConverter for a given RDF library.
+ */
+public interface JellyConverterFactory<
+    TNode,
+    TDatatype,
+    TEncoderConverter extends ProtoEncoderConverter<TNode>,
+    TDecoderConverter extends ProtoDecoderConverter<TNode, TDatatype>
+> {
+    /**
+     * To be implemented by subclasses. Returns an instance of ProtoEncoderConverter for the RDF library.
+     * @since 2.7.0
```

Comment:
I guess the @since declarations are useless, because these ifaces are not the same as in the original anyway... maybe remove all @since in the rewritten module?

Reply:
sure why not